benchmarks/shared/python/data_generators/gnn_cite_graph_gen.py

```diff
@@ -0,0 +1,639 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Create a synthetic graph that is similar to a citation network
+and in OGB format.
+
+Since this matches OGB MAG/MAD240 - the node types and edge types are set
+Node Types:  Paper, Author, Institution   
+Edge Types:  cites, writes, affiliated with
+
+Also, since the format is defined, so are subdirectories:
+
+gived a "base_dir", the following 4 subdirectroies are created
+* paper
+* paper__cites__paper
+* author__affiliated_with__institution
+* author__writes__paper
+
+Lastely, the output files names and types (whats in the files) is also defined
+
+Example:
+python gnn_cite_graph_gen.py \
+    -mg True \
+    -outdir '/tmp/GNN2' \
+    -papersScale 12 \
+    -format "csv" \
+    -papersLabeledPercent 0.10 \
+    -papersFeatureNoise 0.0001
+
+
+"""
+
+
+import os
+import ast
+import argparse
+
+import numpy as np
+import pandas as pd
+
+import rmm
+
+import cugraph
+from cugraph.generators import rmat
+from cugraph.structure import NumberMap
+
+import cudf as cudf_sg
+import dask_cudf as cudf_mg
+import cugraph.dask as dask_cugraph
+
+import dask
+from dask.distributed import Client, wait
+from dask_cuda import LocalCUDACluster
+from cugraph.dask.comms import comms as Comms
+
+
+# --- Some global attributes
+paper_dir = "/paper"
+paper_cite_dir = "/paper__cites__paper"
+author_paper_dir = "/author__writes__paper"
+author_institute_dir = "/author__affiliated_with__institution"
+
+
+def setup_sg() -> None:
+
+    # Set RMM to allocate all memory as managed memory
+    rmm.mr.set_current_device_resource(rmm.mr.ManagedMemoryResource())
+    assert rmm.is_initialized()
+
+
+def setup_mg():
+
+    print("Setup MG")
+
+    cluster = LocalCUDACluster(rmm_managed_memory=True)
+    client = Client(cluster)
+    Comms.initialize(p2p=True)
+
+    return (client, cluster)
+
+
+def stop_mg(client, cluster):
+    Comms.destroy()
+    client.close()
+    cluster.close()
+
+
+def make_dir(d):
+    """
+    create the directory if it does not exists
+    This is a separated funtion in case the directory should be
+    deleted first and recreated (i.e. deleting all existing files)
+    """
+    if not os.path.exists(d):
+        os.mkdir(d)
+
+
+def setup_directories(basedir):
+    """
+    create the 4 directories
+    """
+    # Base
+    make_dir(basedir)
+
+    # Paper
+    make_dir(basedir + paper_dir)
+
+    # Paper Cites
+    make_dir(basedir + paper_cite_dir)
+
+    # Paper Author
+    make_dir(basedir + author_paper_dir)
+
+    # Author - Institution
+    make_dir(basedir + author_institute_dir)
+
+
+def save_data(args, dir, file_name, data, mg=False):
+
+    if mg:
+        f = os.path.join(dir, f"{file_name}")
+    else:
+        f = os.path.join(dir, f"{file_name}.{args.format}")
+    print(f"\tsaving to {f}")
+
+    if format == "parquet":
```

Comment:
```suggestion
    if args.format == "parquet":
```

Reply:
oops.  I changed the input argument and missed this one